xrdsst/controllers/endpoint.py

```diff
@@ -0,0 +1,83 @@
+        self.add_service_endpoints(active_config)
+
+    def add_service_endpoints(self, config):
+        ss_api_conf_tuple = list(zip(config["security_server"], map(lambda ss: self.create_api_config(ss, config), config["security_server"])))
+
+        for security_server, ss_api_config in [t for t in ss_api_conf_tuple if t[1]]:
+            BaseController.log_debug('Starting service description access adding process for security server: ' + security_server['name'])
+            if "clients" in security_server:
+                for client in security_server["clients"]:
+                    if "service_descriptions" in client:
+                        for service_description in client["service_descriptions"]:
+                            for endpoint in service_description["endpoints"]:
+                                self.remote_add_service_endpoints(ss_api_config, security_server, client, service_description, endpoint)
+
+        BaseController.log_keyless_servers(ss_api_conf_tuple)
+
+    def remote_add_service_endpoints(self, ss_api_config, security_server_conf, client_conf, service_description_conf, endpoint_conf):
+        clients_api = ClientsApi(ApiClient(ss_api_config))
+        try:
+            client_controller = ClientController()
+            client = client_controller.find_client(clients_api, security_server_conf, client_conf)
+            if client:
+                try:
+                    service_description = self.get_client_service_description(clients_api, client, service_description_conf)
+                    if service_description:
+                        if service_description.type == ServiceType().WSDL:
+                            BaseController.log_info("Wrong service description, endpoints for WSDL services are not"
+                                                    " allowed, skipped endpoint creation for service '" + service_description.url + "'")
+                        else:
+                            serviceId = service_description.services[0].id
+                            endpoint = Endpoint(id=None, service_code=service_description_conf["rest_service_code"], method=endpoint_conf["method"], path=endpoint_conf["path"], generated= None)
+
+                            try:
+                                services_api = ServicesApi(ApiClient(ss_api_config))
+                                response = services_api.add_endpoint(id=serviceId, body=endpoint)
+                                if response:
+                                    BaseController.log_info("Added service endpoint '" + endpoint.method + " " + endpoint.path + "' for service '" + serviceId + "'")
+                            except ApiException as err:
+                                if err.status == 409:
+                                    BaseController.log_info("Added service endpoint '" + endpoint.method + " " + endpoint.path  + "' for service '" + serviceId  + "' already added")
```

Comment:
Better use here: "BaseController.log_info("Service endpoint '" + endpoint.method + " " + endpoint.path  + "' for service '" + serviceId  + "' already added")"